### PR TITLE
Travel request v2

### PR DESCRIPTION
**Pull request by MugiranezaJ**
![Uploading Screen Shot 2020-11-12 at 12.15.40.png…]
Screen Shot 2020-11-12 at 12 07 40
Screen Shot 2020-11-12 at 12 07 23
Screen Shot 2020-11-12 at 12 05 54
Screen Shot 2020-11-12 at 12 05 40
Screen Shot 2020-11-12 at 12 04 29
Screen Shot 2020-11-12 at 12 04 12
Screen Shot 2020-11-12 at 12 03 48
Screen Shot 2020-11-12 at 12 00 01
Screen Shot 2020-11-12 at 11 57 20
Screen Shot 2020-11-12 at 11 40 59
Screen Shot 2020-11-12 at 11 30 16

Screen Shot 2020-11-12 at 12 15 40

()

### CODE REVIEW

File: src/controllers/travelRequest.controller.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        const request = {
+            managerId:decoded.manager_id,
+            userId:decoded.id,
+            // status:req.body.status,
+            createdAt:new Date(),
+            updatedAt: new Date(),
+        }
+        /*const t = db.sequelize.transaction()*/
+        try{
+            db.TravelRequest.create(request)
```
Comment (salviosage) [inline src/controllers/travelRequest.controller.js]: should we add this in service instead

File: src/helper/tokenToData.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import jwt from 'jsonwebtoken'
+import isUserExist from '../services/findUser';
+export async function getDataFromToken(req, res, next){
+    if(req.headers && req.headers.authorization){
+        const authorization = req.headers.authorization
+        var decoded ='';
+        try {
+            decoded = jwt.verify(authorization, process.env.TOKEN_SECRET);
```
Comment (salviosage) [inline src/helper/tokenToData.js]: should we use the pre created reusable function for this

File: src/models/travelrequest.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    },
+    managerId: {
+      allowNull: false,
+      type:DataTypes.UUID
+    },
+    userId: {
+      allowNull: false,
+      type:DataTypes.UUID
+    },
+    // tripId: {
```
Comment (salviosage) [inline src/models/travelrequest.js]: remove comments please

File: src/models/trip.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+'use strict';
+// const {Model} = require('sequelize');
+module.exports = (sequelize, DataTypes) => {
+//   class Trip extends Model {
+//     /**
+//      * Helper method for defining associations.
+//      * This method is not a part of Sequelize lifecycle.
```
Comment (salviosage) [inline src/models/trip.js]: even here remove comments

**Review by salviosage — COMMENTED** [review]
Nice work , i left some comments please attend them

File: src/helper/tokenToData.js (hunk 2 above)
Comment (MugiranezaJ) [reply src/helper/tokenToData.js]: I used it as suggested, thanks

File: src/models/travelrequest.js (hunk 3 above)
Comment (MugiranezaJ) [reply src/models/travelrequest.js]: I removed them, Thanks

**Review by salviosage — APPROVED** [review]
LGTM

**Pull request merged**